src/cljs/main/broadfcui/common/components.cljs

```diff
@@ -178,7 +179,11 @@
                       :borderRadius 5}
               :data-test-id (:data-test-id props)
               :data-test-state (if disabled? "disabled" "enabled")
-              :onClick (if disabled? (create-error-message disabled?) (:onClick props))}
+              :onClick (if disabled?
+                         (if disabled-handler
+                           (disabled-handler)
+                           (create-error-message disabled?))
+                         (:onClick props))}
```

Comment:
I don't like this approach. Also, if we're mucking with modals we should be converting to the new modal.

Reply:
I will look for the new modal stuff and re-implement.